src/PhpDoc/TypeNodeResolver.php

```diff
@@ -696,7 +698,7 @@ static function (string $variance): TemplateTypeVariance {
 			if (count($genericTypes) === 2) { // iterable<KeyType, ValueType>
 				return new IterableType($genericTypes[0], $genericTypes[1]);
 			}
-		} elseif (in_array($mainTypeName, ['class-string', 'interface-string'], true)) {
+		} elseif (in_array($mainTypeName, ['class-string', 'interface-string', 'enum-string'], true)) {
```

Comment:
Instead of adding this `elseif` clause, another idea is to add another clause.

```diff
-		} elseif (in_array($mainTypeName, ['class-string', 'interface-string'], true)) {
+		} elseif ($mainTypeName === 'enum-string') {
 			if (count($genericTypes) === 1) {
 				$genericType = $genericTypes[0];
-				if ($genericType->isObject()->yes() || $genericType instanceof MixedType) {
+				if ($genericType->isEnum()->yes() || $genericType instanceof MixedType) {
 					return new GenericClassStringType($genericType);
 				}
			}
```

However, Psalm simply equates the two.

* https://github.com/vimeo/psalm/blob/6.3.0/src/Psalm/Internal/Type/TypeParser.php#L755-L758

Reply:
We need to do `TypeCombinator::intersect($genericType, new ObjectType(UnitEnum::class))`. You can then test it with `enum-string<object>` and also `enum-string<SomeInterface>`.